### ISSUE TITLE
allOf is duplicating properties for inherited model

### ISSUE THREAD

**Issue by dalewking** [swagger-api/swagger-core#3049]
This was originally reported [here](https://github.com/swagger-api/swagger-codegen/issues/8792) against swagger-codegen, but I believe it is actually a bug in the parser, so am reporting it here. Also seems related to [this issue](https://github.com/swagger-api/swagger-core/issues/1373).

[Here is a swagger spec](https://api.youneedabudget.com/papi/spec-v1-swagger.json) that demonstrates the problem.

If you look at the definition for some of the Detail types like BudgetDetail, they use allOf to reuse a summary type and add additional properties to it as in:

    "BudgetDetail": {
      "allOf": [
        {
          "$ref": "#/definitions/BudgetSummary"
        },
        {
          "type": "object",
          "properties": { //...

The problem in codegen is that the fields from the summary appear twice, which doesn't compile.

But looking at the parsed version of this file, the duplication has already happened. The above generates a ComposedSchema that looks like this:

```
class ComposedSchema {
    class Schema {
        title: null
        multipleOf: null
        // .... edited for brevity
    }
    allOf: [class Schema {
        title: null
        multipleOf: null
        // .... edited for brevity
    }, class ObjectSchema {
        class Schema {
            title: null
            multipleOf: null
            // .... edited for brevity
            properties: {accounts=class ArraySchema {
```

You can see here that title, multipleOf, and all the properties I removed for brevity (which all came from BudgetSummary) are duplicated in the output of the parser. Since the parser duplicates them codegen duplicates them as well.

**Comment by catbref** [comment]
Possible duplicate of #3030 ?

**Comment by dalewking** [comment]
Good news is that openapi-generator 4.0.0 (which is supposedly going to be released today) does not have this problem.

**Comment by wwang2016** [comment]
Actually the openapi-generator-maven-plugin (3.3.4) reads the correct allOf and generats the code properly.

I do not see the reason for swagger-maven-plugin (v2.0 and above) to generate the non-conforming api definition file, just because swagger-codegen-maven-plugin has issue to parse a openAPI v3 - conforming file. 

As for now, I have to manually correct the generated file before I used it in openapi-generator-maven-plugin. It is very tedious.

**Comment by HugoMario** [comment]
this has been fixed from 3.0.25 versions
